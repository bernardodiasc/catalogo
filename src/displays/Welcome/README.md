# Welcome

## How to use

```
import Welcome from 'displays/Welcome'
```

```
<Welcome />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
