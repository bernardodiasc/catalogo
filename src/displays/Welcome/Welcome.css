.Welcome {
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 100px 0;
}

.Welcome__copy {
  display: block;
}

.Welcome__selector {
  width: 400px;
}
