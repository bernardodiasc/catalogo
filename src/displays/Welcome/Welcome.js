import React, { PureComponent } from 'react'
import './Welcome.css'

import SelectBox from 'displays/SelectBox'

class Welcome extends PureComponent {
  static defaultProps = {
    cities: [],
    onSelect: value => value
  }

  handleChange = (selectedOption) => {
    this.props.onSelect(selectedOption.value)
  }

  render() {
    return (
      <div className="Welcome">
        <div className="Welcome__copy">
          Bem vindo(a) ao melhor cat√°logo comercial do mundo!
        </div>
        <div className="Welcome__selector">
          <SelectBox
            name="select-city"
            placeholder="Selecione uma cidade..."
            onChange={this.handleChange}
            options={this.props.cities}
          />
        </div>
      </div>
    )
  }
}

export default Welcome
