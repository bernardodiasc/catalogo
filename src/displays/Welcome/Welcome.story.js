import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import Docs from 'stories/containers/Docs'

import Welcome from './Welcome'
import WelcomeReadme from './README.md'
storiesOf('Displays/Welcome', module)
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{WelcomeReadme}</Docs>)
  .add('Example', () => <Welcome />)
  .add('Minimal render', () => <Welcome />)
