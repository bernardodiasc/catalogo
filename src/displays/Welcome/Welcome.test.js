import React from 'react'
import { shallow } from 'enzyme'

import Welcome from './Welcome.js'

it('renders without crashing', () => {
  const component = shallow(<Welcome />)
  expect(component).toMatchSnapshot()
})
