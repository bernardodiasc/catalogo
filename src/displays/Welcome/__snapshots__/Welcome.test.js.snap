// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Welcome"
>
  <div
    className="Welcome__copy"
  >
    Bem vindo(a) ao melhor cat√°logo comercial do mundo!
  </div>
  <div
    className="Welcome__selector"
  >
    <SelectBox
      name="select-city"
      onChange={[Function]}
      options={Array []}
      placeholder="Selecione uma cidade..."
    />
  </div>
</div>
`;
